pkg/pipeline-selector/pipeline-selector.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipelineselector
+
+import (
+	"strings"
+
+	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	devfile "github.com/redhat-appstudio/application-service/pkg/devfile"
+	buildappstudiov1alpha1 "github.com/redhat-appstudio/build-service/api/v1alpha1"
+)
+
+// SelectPipelineForComponent evaluates given list of pipeline selectors aginst specified component
+// to find the build pipeline for the component.
+// The first match is returned.
+func SelectPipelineForComponent(component *appstudiov1alpha1.Component, selectors []buildappstudiov1alpha1.InitialBuildPipelineSelector) (*tektonapi.PipelineRef, []tektonapi.Param, error) {
+	selectionParameters, err := getPipelineSelectionParametersForComponent(component)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, selector := range selectors {
+		if buildPipelineRef, buildPipelineAdditionalParams := findMatchingPipeline(selectionParameters, &selector); buildPipelineRef != nil {
```

Comment:
yeah since you only read `selector` in `findMatcingPipeline` at the moment this code scanning complaint is a no-op, but it would be cleaner for use `for i, selector := range selectors` and then use `&selector[i]`